Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/Program.cs

```diff
@@ -1,20 +1,30 @@
-using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Hosting;
+using HealthcareHospitalManagementSystem.Services;
+using HealthcareHospitalManagementSystem.Infrastructure;
 
-namespace HealthcareHospitalManagementSystem
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+builder.Services.AddSingleton<IDoctorService, DoctorService>();
```

Comment:
Must be AddScoped instead of AddSingleton
https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionserviceextensions.addscoped?view=net-8.0